string source
float32 current_left
float32 current_right
float32 target_left
float32 target_right
